### PR TITLE
Remove necessity of repository connection

### PR DESCRIPTION
**Pull request by steuhs**
This allows the tool to be used in repositories other than GitHub

**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: tools/coverage/presubmit_test.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func TestRunPresubmit(t *testing.T) {
 	arts := artsTest.LocalArtsForTest("TestRunPresubmit")
 	arts.ProduceProfileFile("./" + test.CovTargetRelPath)
 	p := preSubmitForTest()
-	RunPresubmit(p, arts)
+	RunPresubmit(p, arts, false)
```
Comment (yt3liu) [inline tools/coverage/presubmit_test.go]: Add unit test for `RunPresubmit(p, arts, true)`

File: tools/coverage/main.go [hunk 2]
```diff
@@ -48,16 +49,17 @@ func main() {
 	artifactsDir := flag.String("artifacts", defaultArtifactsDir, "directory for artifacts")
 	coverageTargetDir := flag.String("cov-target", defaultCoverageTargetDir, "target directory for test coverage")
 	coverageProfileName := flag.String("profile-name", defaultCoverageProfileName, "file name for coverage profile")
+	noRepoConn := flag.Bool("no-repo-connection", true, "enable it when connection to GibHub is not required")
```
Comment (srinivashegde86) [inline tools/coverage/main.go]: why is this true by default? I think this should default to false and we should assume github by default. Also, rename this flag to something like `isGithub`

File: tools/coverage/main.go [hunk 3]
```diff
@@ -48,16 +49,17 @@ func main() {
...
 	githubTokenPath := flag.String("github-token", "", "path to token to access github repo")
-	covThresholdFlag := flag.Int("cov-threshold-percentage", defaultCovThreshold, "token to access github repo")
+	covThresholdFlag := flag.Int("cov-threshold-percentage", defaultCovThreshold, "token to access GibHub repo")
```
Comment (srinivashegde86) [inline tools/coverage/main.go]:
```suggestion
	covThreshold := flag.Int("cov-threshold-percentage", defaultCovThreshold, "token to access GibHub repo")
```

File: tools/coverage/presubmit.go [hunk 4]
```diff
@@ -27,16 +27,19 @@ import (
...
-	concernedFiles := githubUtil.GetConcernedFiles(&p.GithubPr, "")
-
-	if len(*concernedFiles) == 0 {
-		log.Printf("List of concerned committed files is empty, " +
-			"don't need to run coverage profile in presubmit\n")
-		return false
+	var concernedFiles *map[string]bool
+	if !noRepoConn {
+		concernedFiles = githubUtil.GetConcernedFiles(&p.GithubPr, "")
+		if len(*concernedFiles) == 0 {
```
Comment (srinivashegde86) [inline tools/coverage/presubmit.go]:
```suggestion
		if len(githubUtil.GetConcernedFiles(&p.GithubPr, "")) == 0 {
```

Comment (steuhs) [reply tools/coverage/presubmit.go]: not applicable. concernedFiles used later

File: tools/coverage/main.go (hunk 3 above)
Comment (steuhs) [reply tools/coverage/main.go]: addressed in https://github.com/knative/test-infra/pull/1425

File: tools/coverage/main.go (hunk 2 above)
Comment (steuhs) [reply tools/coverage/main.go]: removed in https://github.com/knative/test-infra/pull/1425

File: tools/coverage/calc/delta.go [hunk 5]
```diff
@@ -128,7 +128,7 @@ func (changes *GroupChanges) processChangedFiles(
 	for i, inc := range changes.Changed {
 		pathFromProfile := githubUtil.FilePathProfileToGithub(inc.base.Name())
 		fmt.Printf("checking if this file is in github change list: %s", pathFromProfile)
-		if (*githubFilePaths)[pathFromProfile] == true {
+		if githubFilePaths == nil || (*githubFilePaths)[pathFromProfile] == true {
```
[the remark this answers is not in the document]
Comment (steuhs) [reply tools/coverage/calc/delta.go]: right now the implementation is that we either have github connection or not. We haven't introduced another repository host yet.

**Comment by steuhs** [comment]
closed in favor of https://github.com/knative/test-infra/pull/1425

**Pull request closed without merging**